index.bs

```diff
@@ -3341,6 +3341,219 @@ be the same.
 </div>
 
 
+<h5 id="idl-overloading-vs-union">Overloading vs. union types</h5>
+
+<i>This section is informative.</i>
+
+<div class="non-normative">
+
+For specifications defining IDL [=operations=], it may seem that [=overloaded|overloads=] and a
+combination of [=union types=] and [=optional arguments=] have some feature overlap.
+
+It is first important to note that [=overloaded|overloads=] may have different behaviors than
+[=union types=] or [=optional arguments=], and one <em>cannot</em be fully defined using the other
+(unless, of course, additional prose is provided, which can defeat the purpose of the Web IDL type
+system). For example, consider the {{CanvasDrawPath/stroke()}} operations defined on the
+{{CanvasDrawPath}} interface [[HTML]]:
+
+<pre highlight="webidl">
+    interface CanvasDrawPathExcerpt {
+      void stroke();
+      void stroke(Path2D path);
+    };
+</pre>
+
+Per the ECMAScript language binding, calling <code>stroke(undefined)</code> on a object implementing
+{{CanvasDrawPath}} would attempt to call the second overload, yielding a {{TypeError}} since
+<emu-val>undefined</emu-val> cannot be <a href="#es-to-interface">converted</a> to a {{Path2D}}.
+However, if the operations were instead defined with [=optional arguments=] and merged into one,
+
+<pre highlight="webidl">
+    interface CanvasDrawPathExcerpt {
+      void stroke(optional Path2D path);
+    };
+</pre>
+
+the [=overload resolution algorithm=] would treat the <var ignore>path</var> argument as not present
+given the same ECMAScript language code <code>stroke(undefined)</code>, and not throw any
+exceptions.
```

Comment:
Maybe we should note this latter behavior is actually more like what web developers expect, and that if CanvasDrawPath were defined today, this would use optional arguments instead.